### PR TITLE
[cmake] add interface libraries ot-config-(ftd|mtd|radio)

### PR DESCRIPTION
**Pull request by lmnotran**
This allows for more granular control of any configuration which might need to be incorporated in the core library build

### CODE REVIEW

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Review by romacdon — APPROVED** [review]
LGTM

File: src/core/ftd.cmake [hunk 1]
```diff
@@ -41,6 +41,8 @@ target_include_directories(openthread-ftd PUBLIC ${OT_PUBLIC_INCLUDES} PRIVATE $
 target_sources(openthread-ftd PRIVATE ${COMMON_SOURCES})
 
 target_link_libraries(openthread-ftd
+    PUBLIC
```
Comment (bukepo) [inline src/core/ftd.cmake:44]: does these config have to be public?

Comment (lmnotran) [reply src/core/ftd.cmake:44]: The alternative would be adding `PRIVATE` links to all the other targets that link against the corresponding `-ftd/-mtd/-radio/-rcp` libraries. Is that favorable? I can change the PR to that instead if so

Comment (bukepo) [reply src/core/ftd.cmake:44]: Yes, please. I think it's better not to forward flags if not necessary.

**Comment by lmnotran** [comment]
@bukepo @jwhui Please review when you get a chance

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**